command/agent/dns.go

```diff
@@ -357,6 +358,46 @@ PARSE:
 		query := strings.Join(labels[:n-1], ".")
 		d.preparedQueryLookup(network, datacenter, query, req, resp)
 
+	case "addr":
+		if n < 2 {
```

Comment:
This should probably be `!= 2` so we don't accept any periods, since we know it's just hex stuff.